lib/sugarcrm.js

```diff
@@ -33,314 +32,346 @@ const defaultPlatform = ELASTICIO_TASK_ID && ELASTICIO_STEP_ID
  * @constructor
  */
 module.exports = function Service(config, emitter) {
-    if (!config || !config.baseUrl) {
-        throw new Error('SugarCRM Configuration is missing base URL');
+  if (!config || !config.baseUrl) {
+    throw new Error('SugarCRM Configuration is missing base URL');
+  }
+
+  const cfgUrl = {};
+  const urlParts = urlParser.parse(config.baseUrl);
+
+  if (urlParts.protocol) {
+    cfgUrl.protocol = urlParts.protocol;
+  }
+
+  if (urlParts.hostname) {
+    cfgUrl.baseUrl = urlParts.hostname + urlParts.pathname;
+  } else {
+    cfgUrl.baseUrl = urlParts.pathname;
+  }
+
+  const cfg = { protocol: 'https:', ...cfgUrl, ...config };
+
+  cfg.platform = cfg.platform || defaultPlatform;
+
+  function buildUrl(urlPath) {
+    return `${cfg.protocol}//${path.join(cfg.baseUrl, API_URI, urlPath)}`;
+  }
+
+  /**
+   * Check the response from SugarCRM.  Returns true for 200s responses authenticated.
+   * Returns false if there is a problem with authentication.
+   * Throws an exception for all other response codes.
+   * @returns {boolean}
+   */
+  function wasAuthenticationSuccessful(sugarCrmRepsonse) {
```

Comment:
misspelled, but the same usage in the code, so it is ok :)
`sugarCrmRepsonse` -> `sugarCrmResponse`

Reply:
It was from beginning but we didn't spotted it :). Fixed anyway.